types/index.d.ts

```diff
@@ -5,24 +5,25 @@ import { FastifyPluginCallback, FastifyBaseLogger } from "fastify";
 declare module "fastify" {
   interface FastifyInstance {
     createSecureSession(data?: Record<string, any>): fastifySecureSession.Session
-    decodeSecureSession(cookie: string, log?: FastifyBaseLogger): fastifySecureSession.Session | null
-    encodeSecureSession(session: fastifySecureSession.Session): string
+    decodeSecureSession(cookie: string, log?: FastifyBaseLogger, sessionKey?: string): fastifySecureSession.Session | null
+    encodeSecureSession(session: fastifySecureSession.Session, sessionKey?: string): string
   }
 
   interface FastifyRequest {
     session: fastifySecureSession.Session;
   }
 }
 
-type FastifySecureSession = FastifyPluginCallback<fastifySecureSession.SecureSessionPluginOptions>;
+type FastifySecureSession = FastifyPluginCallback<fastifySecureSession.SecureSessionPluginOptions | (fastifySecureSession.SecureSessionPluginOptions & Required<Pick<fastifySecureSession.SecureSessionPluginOptions, 'sessionKey'>>)[]>;
 
 declare namespace fastifySecureSession {
-  export type Session = Partial<SessionData> & {
+  export type Session<T = SessionData> = Partial<T> & {
     changed: boolean;
     deleted: boolean;
-    get<Key extends keyof SessionData>(key: Key): SessionData[Key] | undefined;
-    set<Key extends keyof SessionData>(key: Key, value: SessionData[Key] | undefined): void;
-    data(): SessionData | undefined;
+    get<Key extends keyof T>(key: Key): T[Key] | undefined;
+    get(key: string): any | undefined;
```

Comment:
This could be dropped for improved type safety. This is only needed to allow the code below to work when overriding the Session type and then getting a key that is not in the interface. 

I don't know if removing would be a breaking change as users choose to override the type + this way was not in the docs before.

```
interface FooSessionData {
  foo: string;
}

declare module "fastify" {
  interface FastifyRequest {
    foo: Session<FooSessionData>;
  }
}

expectType<any>(request.foo.get("baz"));
```